File: agent.js
class AIAgent {
  constructor() {
    this.operations = {
      'add': (x, y) => x + y,
      'subtract': (x, y) => x - y,
      'multiply': (x, y) => x * y,
      'divide': (x, y) => y !== 0 ? x / y : 'Error: Division by zero',
      'sqrt': (x) => x >= 0 ? Math.sqrt(x) : 'Error: Negative number',
      'power': (x, y) => Math.pow(x, y)
    };
  }

  calculate(operation, ...args) {
    if (!this.operations[operation]) {
      throw new Error('Invalid operation');
    }
    return this.operations[operation](...args);
  }
}

module.exports = { AIAgent };

File: agent.py
import math

class AIAgent:
    def __init__(self):
        self.operations = {
            'add': lambda x, y: x + y,
            'subtract': lambda x, y: x - y,
            'multiply': lambda x, y: x * y,
            'divide': lambda x, y: x / y if y != 0 else 'Error: Division by zero',
            'sqrt': lambda x: math.sqrt(x) if x >= 0 else 'Error: Negative number',
            'power': lambda x, y: math.pow(x, y)
        }

    def calculate(self, operation, *args):
        try:
            if operation not in self.operations:
                raise ValueError("Invalid operation")
            return self.operations[operation](*args)
        except Exception as e:
            return str(e)
